deepcell_toolbox/metrics.py

```diff
@@ -810,11 +847,31 @@ def calc_object_stats(self, y_true, y_pred):
             y_pred (numpy.array): Labeled prediction mask
 
         Raises:
-            ValueError: if the shape of the input tensor is less than length three
+            ValueError: If y_true and y_pred are not the same shape
+            ValueError: If data_type is 2D, if input shape does not have ndim 3 or 4
+            ValueError: If data_type is 3D, if input shape does not have ndim 4
         """
 
-        if len(y_true.shape) < 3:
-            raise ValueError('Invalid input dimensions: must be at least 3D tensor')
+        if y_pred.shape != y_true.shape:
+            raise ValueError('Input shapes need to match. Shape of prediction '
+                             'is: {}.  Shape of y_true is: {}'.format(
+                                 y_pred.shape, y_true.shape))
+
+        # If 2D, dimensions can be 3 or 4 (with or without channel dimension)
+        if not self.is_3d:
+            if y_true.ndim not in {3, 4}:
+                raise ValueError('Expected dimensions for y_true (2D data) are 3 or 4.'
+                                 'Accepts: (x, y), (batch, x, y), or (batch, x, y, chan)'
+                                 'Got ndim: {}'.format(y_true.ndim))
```

Comment:
The error makes it sound like `ndim == 2` should be acceptable. Should we support ndim == 2? or just change the error.

Reply:
Comment was erroneous, fixed - ndim == 2 not acceptable